content/client-lib-development-guide/features.textile

```diff
@@ -1025,7 +1025,7 @@ h4. ClientOptions
 ** @(TO3i)@ @recover@ string - A connection recovery string, specified with the intention of inheriting the state of an earlier connection
 ** @(TO3j)@ Auth option attributes:
 *** @(TO3j1)@ @key@ string - Full Ably key string as obtained from dashboard
-*** @(TO3j2)@ @token@ string - An authentication token string issued for this application
+*** @(TO3j2)@ @token@ string | @TokenDetails@ | @TokenRequest@ - An authentication token issued for this application, either in the form of a token string, a @TokenDetails@ object, or a @TokenRequest@ object
```

Comment:
Given Paddy's objection, I guess maybe something like this?
`@(TO3j2)@ @token@ string | @TokenDetails@ | @TokenRequest@ - An authentication token issued for this application, either in the form of a token string, a @TokenDetails@ object, or a @TokenRequest@ object. For languages where it's impractical or unidiomatic to take multiple types, this should take a token string only, and TO3j3 and TO3j12 apply`

`(TO3j3) tokenDetails TokenDetails – for languages where @token@ (TO3j2) takes a string only, this is a @TokenDetails@ object`

`(TO3j12) tokenRequest TokenRequest – for languages where @token@ (TO3j2) takes a string only, this is a @TokenRequest@ object`